app/main.py

```diff
@@ -1,25 +1,73 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, value: str) -> None:
+        self.protected_name = "_" + value
+
+    def __get__(self, instance: Any, owner: Any) -> Any:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: Any, value: Any) -> None:
+        if value in range(self.min_amount, self.max_amount + 1):
+            setattr(instance, self.protected_name, value)
+        else:
+            setattr(instance, self.protected_name, False)
```

Comment:
Here you can raise Exception instead of assigning False to variable

Reply:
nice, will update